extensions/prototype-content-with-issues-table/app.R

```diff
@@ -27,39 +29,51 @@ as_content_list <- function(content_df, client) {
   })
 }
 
-# checks to see if a content item has failed jobs within the last 30d, grabs 
-# usage data if it does, then compiles content, job, and usage data together 
-# into a tibble, returning it.
-get_failed_job_data <- function(item, usage) {
-  failed_jobs <- get_jobs(item) |>
-        # filter out successful and running jobs
-        filter(exit_code != 0 & status != 0 & !(is.na(end_time)))
-  
+# filters content jobs down to failures and sets content_recovered depending on 
+# whether or not the latest job ended in error
+filter_jobs <- function(jobs) {
+  if (is.null(jobs) || nrow(jobs) == 0){
+    # content item does not have any jobs
+    NULL
+  } else {
+    # grab the latest job and all failing jobs
+    latest_job <- jobs %>%
+      slice_max(start_time, with_ties = FALSE)
+    failed_jobs <- jobs %>%
+      # filter out successful and running jobs 
+      filter(exit_code != 0 & !(is.na(exit_code)) & status != 0 & !(is.na(end_time))) |>
+      # grab only the columns we use for cleaner dplyr pipeline
+      select(end_time, exit_code, tag, key) 
+    # set content_recovered depending on if latest_job was in failed_jobs 
+    failed_jobs %>%
+      mutate(
+        content_recovered = ifelse(latest_job$key %in% failed_jobs$key, FALSE, TRUE)
+      )
+  }
```

Comment:
Same comment as https://github.com/posit-dev/connect-extensions/pull/34/files#r2012130724 if you're returning `NULL` when there are no jobs, return early or save both of these branches to a variable and return that one. 

Alternatively, seeing this in a function: would it be possible to have the no-jobs path return a data.frame with zero rows (and make the rest of the code robust to that)? That might be a more natural approach and have fewer of these if/else branches. It also has the nice property that the functions all return _one_ thing (a data.frame with a specific set of columns) rather than two things (either a data.frame or `NULL`, depending). This isn't _required_ in this PR, but something to think about + might clean this code up and make it easier to follow.

Reply:
I replaced the `NULL` returns with an empty dataframe, and updated conditionals to only check that `nrow(df) == 0`.